packages/graphql/src/translate/queryAST/ast/QueryASTContext.ts

```diff
@@ -24,7 +24,7 @@ type Scope = Map<string, Cypher.Variable>;
 
 export class QueryASTEnv {
     private scopes = new Map<Cypher.Node | Cypher.Relationship, Scope>();
-
+    public topLevelOperationName: "READ" | "CREATE" | "UNWIND" = "READ";
```

Comment:
Sure, this is a temporary hack as the full tree is still not available yet, the reason why "UNWIND" was added is because the Read operation is reused for the whole mutation and therefore needs to be collected at the end while in the simple "Create" operation the current behavior is to create a projection for each node created. My plan is to unify most of this discrepency during these refactors but I thought it was better to keep the current behavior for the first iteration! Let's discuss it anyway!